### PR TITLE
Feature#26 deploy app

### LINKED ISSUE #9: Reports view
# REQUIREMENTS
This view shows several reports about the employees and branches information, like attendance by day, average time employees checked in/out by month and absence report by month.

# DESIGN
Provide references to images of the design, how it looks like now, and how it should look like.
INFORMES

# Acceptance Criteria
List all required acceptance criteria
- [x] The end result should look as close as possible to the design
- [x] Show the attendance by day
- [x] Show the average time employees checked in/out by month
- [x] Show the absence report by month

### LINKED ISSUE #26: Deploy of Finish App
# REQUIREMENTS
In this issue the finished app will be deployed on Heroku

# DESIGN
Provide references to images of the design, how it looks like now, and how it should look like.

# Acceptance Criteria
List all required acceptance criteria
- [x] The page must be accessible from any device

### PR DESCRIPTION
**Pull request by FranckZnt19**
We made change to stattic files, since it had production failures
 
when we deployed the project in Heroku, some styles don't load or dont found
## Description: Why did you change/implement that?

changes in some specifications of gemfile and it was deployed in heroku
## Testing: Help me how can I test or look at the changes?

You can see and check this changes in the below link
-[App](https://reloj-checador-team01-kata08.herokuapp.com/)

### CODE REVIEW

File: reloj-checador/Procfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+web: rails server
+release: bundle exec rake db:migrate
```
Comment (YilKanda) [inline reloj-checador/Procfile]: Add a blank line.

File: reloj-checador/Procfile.dev [hunk 2]
```diff
@@ -0,0 +1,2 @@
+web: bundle exec rails server
+webpack: bundle exec bin/webpack-dev-server
```
Comment (YilKanda) [inline reloj-checador/Procfile.dev]: Add a blank line.

File: reloj-checador/app/javascript/packs/layouts/navbar.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        break;
+    case 'companies':
+        document.getElementById("companies_li").classList.add('active');
+        break;
+    case 'employees':
+        document.getElementById("employess_li").classList.add('active');
+        break;
+    default:
+        break;
+}
```
Comment (YilKanda) [inline reloj-checador/app/javascript/packs/layouts/navbar.js]: Add a blank line.

File: reloj-checador/app/javascript/packs/welcome/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+var clock = document.getElementById("clock");
+if (clock != undefined)
+    var renderTime = function () {
+        var time = new Date();
+
+        clock.textContent = `${time.getDate()}-${time.getMonth() + 1}-${time.getFullYear()} ${time.getHours()}:${time.getMinutes()}:${time.getSeconds()}`
+    };
+renderTime();
+setInterval(renderTime, 1000);
```
Comment (YilKanda) [inline reloj-checador/app/javascript/packs/welcome/index.js]: Add a blank line.

File: reloj-checador/app/javascript/stylesheets/application.scss [hunk 5]
```diff
@@ -1 +1,2 @@
-@import "toastr";
+// @import 'toastr/build/toastr.min';
+@import '@fortawesome/fontawesome-free';
```
Comment (YilKanda) [inline reloj-checador/app/javascript/stylesheets/application.scss]: Add a blank line.

File: reloj-checador/app/views/attendaces/_body_table_attendaces.html.erb [hunk 6]
```diff
@@ -5,7 +5,7 @@
     <td><%= attendace.lastname%> </td> 
     <td><%= attendace.position%> </td> 
     <td><%= attendace.date%> </td> 
-    <td><%= attendace.time%> </td>
+    <td><%= attendace&.time&.strftime("%H:%M:%S")%> </td>
     <td><%= attendace.check_type%> </td> 
   </tr>
 <%end%>
```
Comment (YilKanda) [inline reloj-checador/app/views/attendaces/_body_table_attendaces.html.erb]: Add a blank line.

File: reloj-checador/app/views/employees/_body_table_attendaces.html.erb [hunk 7]
```diff
@@ -2,7 +2,7 @@
   <tr>
     <th scope="row"><%= attendace.id%></th>
     <td><%= attendace.date%> </td>
-    <td><%= attendace.time%> </td>
+    <td><%= attendace&.time&.strftime("%H:%M:%S") %> </td>
     <td><%= attendace.check_type%> </td>    
   </tr>
 <%end%>
```
Comment (YilKanda) [inline reloj-checador/app/views/employees/_body_table_attendaces.html.erb]: Add a blank line.

File: reloj-checador/app/views/layouts/_navbar.html.erb [hunk 8]
```diff
@@ -44,20 +44,4 @@
...
-            document.getElementById("companies_li").classList.add('active');
-            break;
-        case 'employees':
-            document.getElementById("employess_li").classList.add('active');
-            break;
-        default:
-            break;
-    }
-</script>
+<%= javascript_pack_tag 'layouts/navbar' %>
```
Comment (YilKanda) [inline reloj-checador/app/views/layouts/_navbar.html.erb]: Add a  blank line

File: reloj-checador/app/javascript/packs/layouts/navbar.js (hunk 3 above)
Comment (YilKanda) [inline reloj-checador/app/javascript/packs/layouts/navbar.js]: WDYT, about simplifying this decision, with an if, like the following:

```
if (controller_page == 'attendaces' || controller_page == 'companies' || controller_page == 'employees' ) {
	document.getElementById(controller_page + "_li").classList.add('active');
}
```

File: reloj-checador/app/views/reports/absence_by_month.html.erb [hunk 9]
```diff
@@ -1,56 +1,56 @@
...
+                    </thead>
+                    <tbody>
+                        <% @abscenes.each do |abscene| %>
+                            <% company = @companies.select{|company| company if company.company_id == abscene.company_id}.first %>
+                            <% abscene_count = company.emp_count - abscene.emp_count %>
+                            <tr>
+                                <td style="text-align: center;"><%=  abscene.date %></td>
+                                <td style="text-align: center;"><%=  company.name %></td>
+                                <td style="text-align: center;"><%=  abscene.emp_count %></td>
+                                <td style="text-align: center;"><%=  (abscene_count > 0) ? abscene_count : "0" %></td>
```
Comment (LDVera) [inline reloj-checador/app/views/reports/absence_by_month.html.erb]: Maybe is better make an sass archive with only an varible like $text-center and use it in all the elements that u need

**Review by LDVera — DISMISSED** [review]
Good job!

File: reloj-checador/app/javascript/packs/layouts/navbar.js (hunk 3 above)
Comment (ramasoftdev) [reply reloj-checador/app/javascript/packs/layouts/navbar.js]: Taking into consideration the sugestion, the code now is:
`if (['attendaces','companies','employees','reports'].indexOf(controller_page) >= 0 ) {
	document.getElementById(controller_page + "_li").classList.add('active');
}`

File: reloj-checador/Procfile (hunk 1 above)
Comment (ramasoftdev) [reply reloj-checador/Procfile]: Added blank line.

File: reloj-checador/Procfile.dev (hunk 2 above)
Comment (ramasoftdev) [reply reloj-checador/Procfile.dev]: Added blank line.

File: reloj-checador/app/javascript/packs/layouts/navbar.js (hunk 3 above)
Comment (ramasoftdev) [reply reloj-checador/app/javascript/packs/layouts/navbar.js]: Code changed. New code is in comments below

File: reloj-checador/app/javascript/packs/welcome/index.js (hunk 4 above)
Comment (ramasoftdev) [reply reloj-checador/app/javascript/packs/welcome/index.js]: Added new line

File: reloj-checador/app/javascript/stylesheets/application.scss (hunk 5 above)
Comment (ramasoftdev) [reply reloj-checador/app/javascript/stylesheets/application.scss]: Added new line and deleted comment line

File: reloj-checador/app/views/attendaces/_body_table_attendaces.html.erb (hunk 6 above)
Comment (ramasoftdev) [reply reloj-checador/app/views/attendaces/_body_table_attendaces.html.erb]: Added new line

File: reloj-checador/app/views/employees/_body_table_attendaces.html.erb (hunk 7 above)
Comment (ramasoftdev) [reply reloj-checador/app/views/employees/_body_table_attendaces.html.erb]: added new line

File: reloj-checador/app/views/layouts/_navbar.html.erb (hunk 8 above)
Comment (ramasoftdev) [reply reloj-checador/app/views/layouts/_navbar.html.erb]: Added blank line.

File: reloj-checador/app/views/attendaces/_form.html.erb [hunk 10]
```diff
@@ -15,11 +15,4 @@
...
 </div>
-
-<script>
-  var clock = document.getElementById("att-new-clock");
-  setInterval(() => {
-    let date = new Date()
-    clock.innerHTML = date.toLocaleTimeString();
-  },1000);
-</script>
+<%= javascript_pack_tag 'attendace/new_attendace' %>
```
Comment (YilKanda) [inline reloj-checador/app/views/attendaces/_form.html.erb]: Add a blank line.

**Review by YilKanda — DISMISSED** [review]
LGTM! Just add a blank line.

**Pull request merged**